### PR TITLE
Gradle 6 compatibility upgrade

### PR DESCRIPTION
**Pull request by GaborGinterAdN**
Also added a default GitHub CI workflow to verify all changes.

Related to Issue #94.

### CODE REVIEW

**Review by adamu — APPROVED** [review]
Looks like this PR will fix the following two warnings:

```
Internal API constructor DefaultPolymorphicDomainObjectContainer(Class<T>, Instantiator) has been deprecated. This is scheduled to be removed in Gradle 7.0.
        at org.gradle.api.internal.DefaultPolymorphicDomainObjectContainer.<init>(DefaultPolymorphicDomainObjectContainer.java:46)
        at org.unbrokendome.gradle.plugins.testsets.dsl.DefaultTestSetContainer.<init>(TestSetContainer.kt:74)
        at org.unbrokendome.gradle.plugins.testsets.dsl.DefaultTestSetContainer_Decorated.<init>(Unknown Source)
```

```
The ProjectLayout.configurableFiles() method has been deprecated. This is scheduled to be removed in Gradle 7.0. Please use the ObjectFactory.fileCollection() method instead.
        at org.gradle.api.internal.file.DefaultProjectLayout.configurableFiles(DefaultProjectLayout.java:99)
        at org.unbrokendome.gradle.plugins.testsets.dsl.DefaultTestSetContainer.createSourceSetForTestSet(TestSetContainer.kt:129)
        at org.unbrokendome.gradle.plugins.testsets.dsl.DefaultTestSetContainer.access$createSourceSetForTestSet(TestSetContainer.kt:71)
        at org.unbrokendome.gradle.plugins.testsets.dsl.DefaultTestSetContainer$entityInstantiator$1.create(TestSetContainer.kt:102)
        at org.unbrokendome.gradle.plugins.testsets.dsl.DefaultTestSetContainer.doCreate(TestSetContainer.kt:144)
        at org.unbrokendome.gradle.plugins.testsets.dsl.DefaultTestSetContainer.doCreate(TestSetContainer.kt:71)
```

👍

File: src/main/kotlin/org/unbrokendome/gradle/plugins/testsets/dsl/TestSetContainer.kt [hunk 1]
```diff
@@ -67,13 +68,13 @@ interface TestSetContainer : PolymorphicDomainObjectContainer<TestSetBase> {
 }
 
 
-@Suppress("DEPRECATION")
 private open class DefaultTestSetContainer
 @Inject constructor(private val project: Project, instantiator: Instantiator)
 // Use this constructor, as they are still supporting it because 'nebula.lint' uses it
```
Comment (sesamzoo) [inline src/main/kotlin/org/unbrokendome/gradle/plugins/testsets/dsl/TestSetContainer.kt:73]: @GaborGinterAdN, might this comment be obsolete, now?

**Comment by sdavids13** [comment]
Any updates on this? I would like to move to Gradle 6 and saw the deprecation warnings which made me a little nervous. Though, you might want to update Gradle to v6.2.2 at this point.

**Comment by tkrullmann** [comment]
I merged it, thanks for the work and sorry for the delay!

**Pull request merged**